web-registry/src/hooks/useBasketDetails.ts

```diff
@@ -1,121 +1,220 @@
+  const [basketClasses, setBasketClasses] = useState<ClassInfo[]>();
+
+  const [batches, setBatches] = useState<string[]>();
+  const basketBatches = useQueryListBatchInfo(batches);
 
-  // fetch all the data
+  const [fetchProjectByBatchDenom] = useProjectByBatchDenomLazyQuery();
+  const [batchesProjects, setBatchesProjects] = useState<BatchWithProject[]>();
+
+  // Batches string list query param for `useQueryListBatchInfo(batches)`
+  useEffect(() => {
+    if (!basketBalances?.balances) return;
+    setBatches(basketBalances.balances.map(balance => balance.batchDenom));
+  }, [basketBalances?.balances]);
+
+  // TODO ? overview data: extract into its own hook / function ?
+  // fetch credit classes metadata
   useEffect(() => {
-    if (basketDenom === undefined) return;
+    if (!basketClassesInfo || basketClassesInfo.length === 0) return;
 
-    async function fetchData(basketDenom: string): Promise<void> {
+    async function fetchData(
+      basketClassesInfo: QueryClassInfoResponse[],
+    ): Promise<void> {
       try {
-        const basket = await getBasket(basketDenom);
-        // TODO: Hardcoded. Fetch basket classes display names
-        const basketClasses = basket.classes.map(classId => ({
-          id: classId,
-          name: `Verified Carbon Standard (${classId})`,
-        }));
-        // TODO: Display basket denom
-        const displayDenom = 'eco.C.rNCT';
-        // TODO: Basket description (see comment in Figma)
-        // https://www.figma.com/file/x5vjWsddiUBzP2N13AFOPw?node-id=32:10028#155844414
-        // const description = getBasketDescription(basketDenom);
-        const description =
-          'The Regen Nature Carbon Ton groups together carbon sequestration ecocredits into one tradeable asset. Lorem ipsum dolor sit amet, consectetur adipiscing elit.';
-        // TODO: Basket Curator
-        const curator = 'regen1qdqafsy2jfuyq7rxzkdwyytmrxlfn8csq0uetx';
-        const balances = await getBasketBalances(basketDenom);
-        const batches = await getBasketBalancesWithBatchInfo(balances.balances);
-
-        setData({
-          basket: basket.basket,
-          basketExtended: {
-            classes: basketClasses,
-            displayDenom,
-            description,
-            curator,
-            totalAmount: 0,
-          },
-          basketBalances: balances.balances,
-          batchesInfo: batches,
-        });
+        const _basketClasses = await Promise.all(
+          basketClassesInfo.map(async basketClass => {
+            // TODO: use metadata
+            // const metadata = await getMetadata(basketClass.info!.metadata);
+            return {
+              id: basketClass.info?.classId || '-',
+              name: basketClass.info?.classId || '-',
+            };
+          }),
+        );
+        setBasketClasses(_basketClasses);
       } catch (error) {
-        // TODO Manage error
         console.error(error); // eslint-disable-line no-console
```

Comment:
could we do some simple error-handling to make sure there aren't endless requests if the ledger can't find something in this file? Like a `[error, setError] = useState(false)`  and then check for `error` in each `useEffect`?

Reply:
Done! (See above)